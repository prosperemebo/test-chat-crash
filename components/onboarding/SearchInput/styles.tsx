import styled from 'styled-components/native'
import { Animated } from 'react-native'
import AntIcon from '@expo/vector-icons/AntDesign'

export const Container = styled(Animated.View)`
  flex-direction: row;
  align-items: center;
  background-color: ${props => props.theme.common.inputBackgroundColor};
  border-radius: 12px;
  margin: 10px 21px;
  height: 46px;
  padding-horizontal: 15px;
  border-width: 1px;
  border-color: ${props => props.theme.common.inputBorderColor};
`

export const IconWrapper = styled.View`
  justify-content: center;
  align-items: center;
  margin-right: 10px;
`

export const SearchIcon = styled(AntIcon)`
  font-size: 20px;
  color: ${props => props.theme.common.inputIconColor};
`

export const ClearIcon = styled(AntIcon)`
  font-size: 18px;
  color: ${props => props.theme.common.inputIconColor};
  margin-left: 10px;
  opacity: 0.7;
`

export const StyledTextInput = styled.TextInput`
  flex: 1;
  color: ${props => props.theme.common.inputTextColor};
  font-family: 'Poppins-Regular';
  font-size: 14px;
  padding: 0;
`