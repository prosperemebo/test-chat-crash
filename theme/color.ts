export const WHITE = '#FFFFFF'
export const t10WHITE = 'rgba(255,255,255,0.1)'
export const t76WHITE = 'rgba(255,255,255,0.76)'

export const RED = '#f60004'
export const RED1 = '#D33E41'


export const BLACK = '#000000'
export const BLACK2 = '#050608'
export const nBLACK = '#15161A'
export const nBLACK10 = '#1F2025'
export const nBLACK30 = '#28292E'
export const nBLACK40 = '#050608'

export const nGRAY10 = '#6B6E79'
export const nGRAY20 = '#858995'
export const nGRAY25 = '#54575e'
export const nGRAY30 = '#2f3137'
export const nGRAY35 = '#52555D'
export const nGRAY40 = '#35363B'


export const tGRAY20 = '#85899599'
export const tGRAY30 = '#545458A6'
export const tGRAY40 = '#45464C'
export const GRAY40 = '#cfcfd0'
export const GRAY45 = '#c9c9cb'
export const GRAY50 = '#aaaaaa'
export const GRAY60 = '#52545D'
export const t40GRAY60 = '#52545D66'
export const GRAY70 = '#383B40'
export const GRAY90 = '#27292d'
export const GRAY100 = '#1f2023'
export const GRAY110 = '#151619'

export const GRAY_ONBOARD = '#8791a5'

export const BLUE = '#50b5e8'
export const BLUE2 = '#0A84FF'

export const PURPLE = '#9A1F99'

export const nBLUE = '#8D88A8'
export const nPURPLE = '#C47AF5'
export const nPURPLE10 = '#873FBB'
export const nPURPLE20 = '#873FBB'
export const nPURPLE40 = '#C882F8'

export const nORANGE10 = '#FF6705'
export const nORANGE20 = '#FF7905'
export const nORANGE40 = '#FFB700'

export const YELLOW2 = '#EBC789'
export const nYELLOW = '#F5D193'

export const nGRAYBACKGROUND2 = '#292A30'
export const nGRAYBACKGROUND3 = '#383940'
export const nGRAYBACKGROUND4 = '#989898'

export const gradient = {
  COLORS: [BLUE, PURPLE],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}

export const orangeGradient = {
  COLORS: [nORANGE10, nORANGE40],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}