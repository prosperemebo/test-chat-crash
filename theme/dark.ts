import {
  WHITE,
  BLACK,
  nGRAY10,
  GRAY_ONBOARD,
  nPURPLE,
  nBLUE,
  GRAY40,
  nGRAY20,
  RED1,
  tGRAY20,
  RED,
  nBLACK,
  nORANGE10,
  nORANGE40,
  nPURPLE10,
  GRAY70,
  tGRAY30,
  BLUE2,
  BLACK2,
  nGRAY25,
  nBLACK30,
  GRAY50,
  nBLACK10,
  t10WHITE,
  nGRAYBACKGROUND2,
  GRAY100,
  GRAY60,
  nGRAY40,
  nGRAY30,
  YELLOW2,
  nYELLOW,
  nGRAYBACKGROUND3,
  tGRAY40,
  nGRAY35,
  nBLACK40,
  t40GRAY60,
  nPURPLE20,
  nPURPLE40,
} from './color'

export const dark = {
  tint: nPURPLE,
  primaryTextColor: WHITE,
  primaryBg: nBLACK10,
  headerBg: GRAY100,
  headerBorderColor: GRAY60,
  secondaryTextColor: GRAY60,
  secondaryBgLight: nBLACK30,
  secondaryBg: nBLACK,
  serviceBg: GRAY100,
  inActiveBorderColor2: t10WHITE,
  backgroundColor2: nGRAYBACKGROUND2,
  backgroundColor3: nGRAYBACKGROUND3,
  backgroundColor4: nBLACK10,

  auth: {
    socialIconColor: WHITE,
    socialLoginButton: BLACK,
    dividerColor: nGRAY10,
    noticeText: WHITE,
    forgotPasswordLabel: WHITE,
  },

  onboarding: {
    headerTextColor: WHITE,
    headerStepsTextColor: GRAY_ONBOARD,
    progressBarColor: nPURPLE,
    progressBarBackgroundColor: nBLUE,
    selectedItemTextColor: WHITE,
    disabledItemTextColor: GRAY40,
    gameSectionOverlayColor: nPURPLE,
    gameEmptySectionColor: BLACK,
    gameEmptyIconColor: BLACK2,
  },

  common: {
    labelColor: WHITE,
    infoTextColor: nGRAY20,
    inputIconColor: nGRAY20,
    inputPlaceholderColor: nGRAY20,
    serviceTextColor: WHITE,
    inputTextColor: WHITE,
    inputTextError: RED1,
    inputBorderColor: tGRAY20,
    inputErrorBorderColor: RED,
    inputBackgroundColor: nBLACK,
    toastBackground: nBLACK,
    toastBorderColor: nPURPLE,
    toastTextColor: WHITE,
    buttonGradientOrangeStart: nORANGE10,
    buttonGradientOrangeEnd: nORANGE40,
    buttonGradientPurpleStart: nPURPLE40,
    buttonGradientPurpleEnd: nPURPLE20,
    buttonGradientGrayStart: nGRAY10,
    buttonGradientGrayEnd: nGRAY20,
    buttonDartBackgroundColor: BLACK,
    backButtonColor: nGRAY20,
    switchTextColor: WHITE,
    checkboxIconColor: nPURPLE,
    gradientButtonColorStart: nPURPLE10,
    gradientButtonColorEnd: nPURPLE,
    borderColor: GRAY70,
    genderPickerBorderColor: nGRAY25,
    genderPickerActiveTextColor: WHITE,
    genderPickerTextColor: GRAY50,
    checkboxTextColor: WHITE,
    iconButtonColor: WHITE,
    backgroundColor: nBLACK10,
  },

  settings: {
    socialIconBackgroundColor: nGRAY40,
    premiumFeatureTextColor: nPURPLE,
    buttonIconColor: nGRAY20,
    serviceItemBgColor: nGRAY30,
  },

  postOnboarding: {
    backgroundColor: nBLACK10,
    inputIconColor: nGRAY20,
    inputIconErrorColor: RED1,
    disclosureTextColor: nGRAY20,
    loadingIndicatorColor: BLACK,
  },

  profile: {
    backgroundColor: nBLACK10,
    improveBackgroundColor: YELLOW2,
    improveBannerBackgroundColor: nYELLOW,
    improveBannerTitleColor: nBLACK30,
    coverPictureCardRemoveButtonBg: tGRAY30,
    profilePicturesCountColor: tGRAY40,
  },

  chats: {
    iconColor: nGRAY20,
    inputColor: nBLACK40,
    placeholderColor: nGRAY35,
    appbarBorderColor: t40GRAY60,
    backgroundColor: nBLACK10,
  },

  nav: {
    bottomTabLabelBackgroundColor: nBLACK,
    bottomTabBackgroundColor: nBLACK,
  },

  other: {
    updateApp: {
      textBorderColor: tGRAY30,
      buttonTextColor: BLUE2,
      textColor: WHITE,
    },
  },
}
