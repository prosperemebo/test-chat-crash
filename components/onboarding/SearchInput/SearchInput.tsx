import { useRef, useEffect } from 'react'
import { Animated, TouchableOpacity } from 'react-native'
import { Container, IconWrapper, SearchIcon, StyledTextInput, ClearIcon } from './styles'

interface Props {
  value: string
  onChangeText: (text: string) => void
  placeholder?: string
}

const SearchInput = ({ value, onChangeText, placeholder }: Props) => {
  const animatedValue = useRef(new Animated.Value(0)).current

  useEffect(() => {
    Animated.timing(animatedValue, {
      toValue: 1,
      duration: 300,
      useNativeDriver: true,
    }).start()
  }, [])

  const handleClear = () => {
    onChangeText('')
  }

  return (
    <Container style={{ opacity: animatedValue }}>
      <IconWrapper>
        <SearchIcon name="search1" />
      </IconWrapper>
      <StyledTextInput
        testID="search-input"
        value={value}
        onChangeText={onChangeText}
        placeholder={placeholder}
        placeholderTextColor="#8791a5"
        autoCapitalize="none"
      />
      {value.length > 0 && (
        <TouchableOpacity testID="clear-button" onPress={handleClear}>
          <ClearIcon name="close" />
        </TouchableOpacity>
      )}
    </Container>
  )
}

export default SearchInput