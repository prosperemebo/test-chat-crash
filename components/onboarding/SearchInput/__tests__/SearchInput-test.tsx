import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { describe, expect, it, jest, beforeEach } from '@jest/globals'
import SearchInput from '../SearchInput'
import { dark } from '@/theme'

const mockOnChangeText = jest.fn()

const renderComponent = (props = {}) => {
  return render(
    <ThemeProvider theme={dark}>
      <SearchInput
        value=""
        onChangeText={mockOnChangeText}
        placeholder="Search games"
        {...props}
      />
    </ThemeProvider>,
  )
}

describe('<SearchInput />', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('renders correctly', () => {
    const { toJSON } = renderComponent()
    expect(toJSON()).toMatchSnapshot()
  })

  it('shows clear button when text is entered', () => {
    const { getByTestId, queryByTestId } = renderComponent({ value: 'test' })
    
    expect(getByTestId('clear-button')).toBeTruthy()
    expect(queryByTestId('clear-button')).not.toBeNull()
  })

  it('hides clear button when no text', () => {
    const { queryByTestId } = renderComponent({ value: '' })
    
    expect(queryByTestId('clear-button')).toBeNull()
  })

  it('calls onChangeText when clear button is pressed', () => {
    const { getByTestId } = renderComponent({ value: 'test' })
    
    fireEvent.press(getByTestId('clear-button'))
    expect(mockOnChangeText).toHaveBeenCalledWith('')
  })

  it('calls onChangeText when text is entered', () => {
    const { getByTestId } = renderComponent()
    
    fireEvent.changeText(getByTestId('search-input'), 'test')
    expect(mockOnChangeText).toHaveBeenCalledWith('test')
  })
})