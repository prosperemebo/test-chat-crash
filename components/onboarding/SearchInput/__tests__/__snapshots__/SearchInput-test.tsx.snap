// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> renders correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#15161A",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderColor": "#85899599",
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 46,
      "marginBottom": 10,
      "marginLeft": 21,
      "marginRight": 21,
      "marginTop": 10,
      "opacity": 0,
      "paddingHorizontal": 15,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "marginRight": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 12,
          },
          {
            "color": "#858995",
            "fontSize": 20,
          },
          {
            "fontFamily": undefined,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      î™°
    </Text>
  </View>
  <TextInput
    autoCapitalize="none"
    onChangeText={[MockFunction]}
    placeholder="Search games"
    placeholderTextColor="#8791a5"
    style={
      {
        "color": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "fontFamily": "Poppins-Regular",
        "fontSize": 14,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      }
    }
    testID="search-input"
    value=""
  />
</View>
`;
